src/meta/expr.lean

```diff
@@ -372,6 +372,44 @@ e.replace $ λ e d,
   | _ := none
   end
 
+/-- FIXME err, what is this called, and is there just a punctuation incantation I can use? -/
+def foo {α : Type*}{m : Type* → Type*} [monad m] (x : m (option α)) (y : m α) : m α :=
+do o ← x,
+  match o with
+  | none := y
+  | (some a) := pure a
+  end
```

Comment:
If you want to up your punctuation,
```suggestion
x >>= λ o, option.rec y pure o
```

Reply:
Perhaps the more general statement you want is
```
def option.mrec {α β : Type*} {m : Type* → Type*} [monad m] (x : m (option α)) (y : m β) (z : α → m β) : m β :=
x >>= @option.rec _ (λ _, m β) y z
```

Reply:
I've add this, and `option.mget_or_else`, that just substitutes in `pure` for `z`.